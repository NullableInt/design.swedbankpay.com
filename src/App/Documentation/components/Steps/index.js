import React, { Component } from "react";
import PrismCode from "react-prism";

import { ComponentPreview, DocToc } from "#";
import StepsComponent from "@/Steps";

const BasicSteps = () => {
    const steps = [
        {
            title: "Step one",
            subtitle: "24.12.17 12:10",
            completed: true

        },
        {
            title: "Step two",
            ongoing: true,
            selected: true
        },
        {
            title: "Step three"
        },
        {
            title: "Step four"
        }
    ];

    return (
    <>
        <h2 id="basic-usage">Basic usage</h2>
        <p>Add class <code className="token property">.steps</code> to an <PrismCode className="language-html">{"<ol>"}</PrismCode> element.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <StepsComponent steps={steps} />
        </ComponentPreview>
    </>
    );
};

const VerticalSteps = () => {
    const steps = [
        {
            title: "Step one",
            completed: true
        },
        {
            title: "Step two",
            ongoing: true,
            selected: true
        },
        {
            title: "Step three"
        },
        {
            title: "Step four"
        }
    ];

    return (
    <>
        <h2 id="vertical-steps">Vertical steps</h2>
        <p>In addition to class <code className="token property">.steps</code>, add the extension class <code className="token property">.steps-vertical</code> to an <PrismCode className="language-html">{"<ol>"}</PrismCode> element.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <StepsComponent steps={steps} vertical />
        </ComponentPreview>
    </>
    );
};

const ClickableSteps = () => {
    const steps = [
        {
            title: "Step one",
            completed: true,
            clickable: true
        },
        {
            title: "Step two",
            ongoing: true,
            clickable: true
        },
        {
            title: "Step three",
            clickable: true
        },
        {
            title: "Step four",
            selected: true
        },
        {
            title: "Step Five"
        },
        {
            title: "Step Six"
        }
    ];

    return (
    <>
        <h2 id="clickable-steps">Clickable steps</h2>
        <p>
            Add an anchor element (<PrismCode className="language-html">{"<a>"}</PrismCode>) to the list item element ( <PrismCode className="language-html">{"<li>"}</PrismCode>).
            Ensure that all content within the list element is also inside the anchor element.
            <br/>Clickable steps is not working properly
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <StepsComponent steps={steps} />
        </ComponentPreview>
    </>
    );
};

const StepsText = () => (
    <div className="col-12 col-lg-10 doc-body">
        <p className="lead">Steps.</p>
        <BasicSteps />
        <VerticalSteps />
        <ClickableSteps />
    </div>
);

class Steps extends Component {
    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <StepsText />
                    <DocToc component={StepsText} />
                </div>
            </div>
        );
    }
}

export default Steps;

/* for testing */
export { BasicSteps, VerticalSteps, ClickableSteps, StepsText };
