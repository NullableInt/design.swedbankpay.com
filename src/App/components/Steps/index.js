import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

const DefaultSteps = [
    {
        title: "Step one",
        subtitle: "24.12.17 12:10",
        completed: true
    },
    {
        title: "Step two",
        ongoing: true
    },
    {
        title: "Step three",
        selected: true
    },
    {
        title: "Step four"
    }
];


const StepContent = ({completed, subtitle, title}) => ( 
    <>
        {completed ? <div className="material-icons steps-icon" >check</div> : null}
        {title}
        {subtitle ? <div className="steps-sub-title">{subtitle}</div> : null}
    </>
);

const RenderSteps = ({ steps }) => (
    <>
        {steps.map(({ title, subtitle, completed, ongoing, selected, clickable }, i) => (      
            <li key={i} className={classnames(completed ? "steps-completed" : null,
                ongoing ? "steps-ongoing" : null,
                selected ? "steps-selected" : null)}>
                {console.log(steps) }
                {clickable ? <a><StepContent completed={completed} subtitle={subtitle} title={title}/></a> : <StepContent completed={completed} subtitle={subtitle} title={title}/> }

            </li>
        ))}
    </>
);

const Steps = ({ steps, vertical }) => {
    const stepsClasses = classnames(
        "steps",
        vertical ? "steps-vertical" : null,
    );

    return (<ol className={stepsClasses}>
        {steps ? <RenderSteps steps={steps} /> : <RenderSteps steps={DefaultSteps} />}
    </ol>
    );
};

Steps.propTypes = {
    steps: PropTypes.array,
    vertical: PropTypes.bool,
    clickable: PropTypes.bool
};

export default Steps;

// exporting for testing purposes
export { DefaultSteps };
