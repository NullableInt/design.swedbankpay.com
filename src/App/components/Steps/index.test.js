import React from "react";
import { shallow } from "enzyme";

import Steps from "./index";

describe("Component: Steps", () => {
    it("is defined", () => {
        expect(Steps).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(
            <Steps />
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.html()).toContain("steps");
    });
});
