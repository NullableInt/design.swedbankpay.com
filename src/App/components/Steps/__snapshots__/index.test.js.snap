// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Steps renders 1`] = `
<ol
  className="steps"
>
  <RenderSteps
    steps={
      Array [
        Object {
          "completed": true,
          "subtitle": "24.12.17 12:10",
          "title": "Step one",
        },
        Object {
          "ongoing": true,
          "title": "Step two",
        },
        Object {
          "selected": true,
          "title": "Step three",
        },
        Object {
          "title": "Step four",
        },
      ]
    }
  />
</ol>
`;
